mesonbuild/envconfig.py

```diff
@@ -291,78 +291,92 @@ def from_literal(cls, literal: T.Dict[str, str]) -> 'MachineInfo':
 
         return cls(system, cpu_family, literal['cpu'], endian, kernel, subsystem)
 
+    @lazy_property
     def is_windows(self) -> bool:
         """
         Machine is windows?
         """
         return self.system == 'windows'
```

Comment:
In particular, all the envconfig functions literally translate a string comparison against a single property to a named function.

Reply:
> We could also just make this a global constant just like _in_ci, no need to make it specifically be a class attribute on the logger.

> Same with all the rest, really.

Here, it didn't used to be a global constant, and the point is just that if we're going to convert it from a method to a property (lazy or not) we might as well make it a precomputed property

I'm assuming the string comparison doesn't show up in profiling, but tons of function calls do...

Reply:
It is better like that? If so, I will just squash the 2 commits.